### ISSUE TITLE
[a11y]: DatePicker [Range With Calendar] - year edit spinbox keyboard bug

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#15253]
### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.39.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Description
not keyboard navigable

The control is not keyboard navigable with tab/shift tab keys.

### Recommended Fix
Make the control keyboard navigable with tab/shift tab keys

When the standard keyboard navigation method is not supported, it impacts the user’s ability to interact with page controls. This affects screen reader and keyboard-only users. 

### WCAG 2.1 Violation
2.1.1 Keyboard

**Comment by heloiselui** [comment]
Fixed, the issue was retested by our accessibility team (in @carbon/react v1.84). The spin box control is not available on screen. Calendar widget appears on focus and is accessible.
